tests/unit/test_hyper_transformer.py

```diff
@@ -1,101 +1,676 @@
+        float_transformer = Mock()
+
+        field_transformers = {
+            'integer': int_transformer,
+            'float': float_transformer,
+            ('integer',): int_transformer
+        }
+        ht = HyperTransformer()
+        ht.field_transformers = field_transformers
+
+        # Run / Assert
+        with pytest.raises(ValueError):
+            ht._validate_field_transformers()
+
+    @patch('rdt.hyper_transformer.HyperTransformer._create_multi_column_fields')
+    @patch('rdt.hyper_transformer.HyperTransformer._validate_field_transformers')
+    def test___init__(self, validation_mock, multi_column_mock):
         """Test create new instance of HyperTransformer"""
         # Run
         ht = HyperTransformer()
 
         # Asserts
         self.assertTrue(ht.copy)
-        self.assertEqual(ht.dtypes, None)
+        self.assertEqual(ht.field_transformers, {})
+        self.assertEqual(ht.data_type_transformers, {})
+        self.assertEqual(ht.field_types, {})
+        multi_column_mock.assert_called_once()
+        validation_mock.assert_called_once()
+
+    def test__create_multi_column_fields(self):
+        """Test the ``_create_multi_column_fields`` method.
+
+        This tests that the method goes through both the ``field_transformers``
+        dict and ``field_types`` dict to find multi_column fields and map
+        each column to its corresponding tuple.
+
+        Setup:
+            - instance.field_transformers will be populated with multi-column fields
+            - instance.field_types will be populated with multi-column fields
+
+        Output:
+            - A dict mapping each column name that is part of a multi-column
+            field to the tuple of columns in the field it belongs to.
+        """
+        # Setup
+        ht = HyperTransformer()
+        ht.field_transformers = {
+            'a': BooleanTransformer,
+            'b': DatetimeTransformer,
+            ('c', 'd'): DatetimeTransformer,
+            'e': NumericalTransformer
+        }
+        ht.field_types = {
+            'f': 'categorical',
+            ('g', 'h'): 'datetime'
+        }
+
+        # Run
+        multi_column_fields = ht._create_multi_column_fields()
+
+        # Assert
+        expected = {
+            'c': ('c', 'd'),
+            'd': ('c', 'd'),
+            'g': ('g', 'h'),
+            'h': ('g', 'h')
+        }
+        self.assertEqual(multi_column_fields, expected)
 
-    def test__analyze(self):
-        """Test _analyze"""
+    def test__get_next_transformer_field_transformer(self):
+        """Test the ``_get_next_transformer method.
+
+        This tests that if the transformer is defined in the
+        ``instance.field_transformers`` dict, then it is returned
+        even if the output type is final.
+
+        Setup:
+            - field_transformers is given a transformer for the
+            output field.
+            - field_types will be given a different transformer
```

Comment:
This test (and the ones below) mention `field_types` but I believe this is `data_type_transformers`